tests/validation/tests/v3_api/test_encryption_secrets.py

```diff
@@ -0,0 +1,48 @@
+from .common import *  # NOQA
+import pytest
+from .test_rke_cluster_provisioning import rke_config
+
+providers = [{"aescbc": {"keys": [{"name": "k-fw5hn", "secret": "RTczRjFDODMwQzAyMDVBREU4NDJBMUZFNDhCNzM5N0I="}]}}, {"identity": {}}]
+custom_resources = [{"resources": ["secrets"], "providers": providers}]
+custom_config = {"apiVersion": "apiserver.config.k8s.io/v1", "kind": "EncryptionConfiguration", "resources": custom_resources}
+SECRET_ENCRYPTION_CONFIG = {"customConfig": custom_config, "enabled": True, "type": "/v3/schemas/secretsEncryptionConfig"}
+
+def test_create_custom_cluster_with_encryption():
+    # Creates custom cluster
+    aws_nodes = \
+        AmazonWebServices().create_multiple_nodes(
+            3, random_test_name("jenkins-custom-secret-encryption"))
+    node_roles = [["controlplane"], ["etcd"],
+                  ["worker"]]
+    client = rancher.Client(url=CATTLE_TEST_URL + "/v3",
+                            token=USER_TOKEN, verify=False)
+    cluster = client.create_cluster(
+        name=random_name(),
+        driver="rancherKubernetesEngine",
+        rancherKubernetesEngineConfig=rke_config)
+    assert cluster.state == "provisioning"
+    i = 0
+    for aws_node in aws_nodes:
+        docker_run_cmd = \
+            get_custom_host_registration_cmd(
+                client, cluster, node_roles[i], aws_node)
+        aws_node.execute_command(docker_run_cmd)
+        i += 1
+    cluster = validate_cluster_state(client, cluster)
+    cluster_rke_config = cluster.rancherKubernetesEngineConfig
+    print(cluster_rke_config)
+    # Adds custom encryption to the RKE config
+    custom_encryption_rke_config = cluster_rke_config.copy()
+    custom_encryption_rke_config["services"]["kubeApi"].update({"secretsEncryptionConfig":SECRET_ENCRYPTION_CONFIG})
+    print(custom_encryption_rke_config)
+    cluster = client.update(cluster,
+                            name=cluster.name,
+                            rancherKubernetesEngineConfig=custom_encryption_rke_config)
+    validate_cluster(client, cluster, intermediate_state="updating")
+
+    # Checks for key in encryption configuration
+    configuration_file = aws_nodes[0].execute_command("sudo cat /etc/kubernetes/ssl/encryption.yaml")
+    print(configuration_file)
+    assert "aescbc" in configuration_file[0]
+    assert "k-fw5hn" in configuration_file[0]
+    assert "RTczRjFDODMwQzAyMDVBREU4NDJBMUZFNDhCNzM5N0I=" in configuration_file[0]
```

Comment:
Is there any other way to assert that encryption has been set on the cluster?

Reply:
Added a check to create and validate secrets.